src/rubrix/client/datasets.py

```diff
@@ -763,7 +763,9 @@ def from_pandas(
         return super().from_pandas(dataframe)
 
     @_requires_datasets
-    def prepare_for_training(self) -> "datasets.Dataset":
+    def prepare_for_training(
+        self, framework="transformers"
```

Comment:
Here I would use an `Enum` type and move the error message of not supported frameworks there. See the `SortBy` class for an example of this pattern: 
https://github.com/recognai/rubrix/blob/fdc851106b204683b4e144fc238267b8e3f8be48/src/rubrix/labeling/text_classification/label_errors.py#L28 

and how to use it as an argument later on:  
https://github.com/recognai/rubrix/blob/fdc851106b204683b4e144fc238267b8e3f8be48/src/rubrix/labeling/text_classification/label_errors.py#L44

Reply:
Hmm, I think I'm not sure what you mean. Where should I move the error message if it's not in the function? In the SortBy class there is a method that raises a not found error, so it would be in two places now.

Reply:
Arg sorry, maybe I was a bit too quick here. What I meant is something like this

```python
# create a Enum-type Framework class like the `SortBy` one
class Framework(Enum):
    TRANSFORMERS = "transformers"
    ...
    
# then use it in the method:
def prepare_for_training(self, framework: Union[Framework, str] = "transformers"):
    # turn str into a Framework instance and trigger error if str is not valid
    if isinstance(framework, str):   
        framework = Framework(framework)
    
    if framework is Framework.TRANSFORMERS:
        ...
```

Reply:
:ok_hand: perfect!